bugbug/models/component.py

```diff
@@ -201,5 +187,152 @@ def is_meaningful(product, component):
+                success = False
+
+            elif bugs_number[full_comp] <= 0:
+                msg = f"Component {component!r} of product {product!r} have 0 bugs or less in it, failure"
+                print(msg)
+                success = False
+
+        # Check number 2, check that conflated components in
+        # CONFLATED_COMPONENTS match at least one component which has more
+        # than 0 bugs
+
+        for conflated_component in self.CONFLATED_COMPONENTS:
+
+            matching_components = [
+                full_comp
+                for full_comp in bugs_number.keys()
+                if full_comp.startswith(conflated_component)
+            ]
+
+            if not matching_components:
+                msg = f"{conflated_component} doesn't match any component"
+                print(msg)
+                success = False
+                continue
+
+            matching_components_values = [
+                bugs_number[full_comp]
+                for full_comp in matching_components
+                if bugs_number[full_comp] > 0
+            ]
+
+            if not matching_components_values:
+                msg = f"{conflated_component} should match at least one component with more than 0 bugs"
+                print(msg)
+                success = False
+
+        # Check number 3, check that values of CONFLATED_COMPONENTS_MAPPING
+        # still exist as components and have more than 0 bugs
+
+        for full_comp in self.CONFLATED_COMPONENTS_MAPPING.values():
+
+            if full_comp not in bugs_number:
+                msg = f"{full_comp} from conflated component mapping doesn't exists, failure"
+                print(msg)
+                success = False
+            elif bugs_number[full_comp] <= 0:
+                msg = f"{full_comp} from conflated component mapping have less than 1 bug, failure"
+                print(msg)
+                success = False
+
+        # Check number 4, conflated components in CONFLATED_COMPONENTS either
+        # exist as components or are in CONFLATED_COMPONENTS_MAPPING
+
+        for conflated_component in self.CONFLATED_COMPONENTS:
+
+            in_mapping = conflated_component in self.CONFLATED_COMPONENTS_MAPPING
+
+            matching_components = [
+                full_comp
+                for full_comp in bugs_number.keys()
+                if full_comp.startswith(conflated_component)
+            ]
+
+            if not (matching_components or in_mapping):
+                msg = f"It should be possible to maps {conflated_component}"
+                print(msg)
+                success = False
+                continue
+
+        # Check number 5, there is no component with many bugs that is not in
+        # meaningful_product_components
+
+        # Recompute the meaningful components
+
+        def generate_meaningfull_tuples():
+            for full_comp, count in bugs_number.items():
+                product, component = full_comp.split("::", 1)
```

Comment:
Nit: to avoid the split, you could make get_product_component_csv_report return tuples instead of full_comp

Reply:
Yeah, but it's slightly easier to check if a component from `CONFLATED_COMPONENTS` match an existing component if the map keys are full components